pytac/lattice.py

```diff
@@ -215,17 +259,18 @@ def get_length(self):
         Returns:
             float: The length of the lattice (m).
         """
-        total_length = 0
+        total_length = 0.0
         for e in self._elements:
             total_length += e.length
         return total_length
 
     def add_element(self, element):
-        """Append an element to the lattice.
+        """Append an element to the lattice and update its lattice reference.
 
         Args:
             element (Element): element to append.
         """
+        element._lattice = self
```

Comment:
This is setting to a private attribute of the element. We should have a `set_lattice()` or `add_to_lattice()` method.

Reply:
added
```
    def set_lattice(self, lattice):
        """Set the stored lattice reference for this element to the passed
        lattice object.

        Args:
            lattice (Lattice): lattice object to store a reference to.
        """
        self._lattice = lattice
```